pina/problem/abstract_problem.py

```diff
@@ -41,26 +38,33 @@ def __init__(self):
                     self.domains[cond_name] = cond.domain
                     cond.domain = cond_name
 
+        self._collect_data = {}
+
     @property
-    def batching_dimension(self):
+    def collected_data(self):
         """
-        Get batching dimension.
+        Return the collected data from the problem's conditions.
 
-        :return: The batching dimension.
-        :rtype: int
+        :return: The collected data.
```

Comment:
Maybe here specify the form of the dictionary, otherwise is not really clear how to use it.